master/internal/api_task.go

```diff
@@ -0,0 +1,27 @@
+package internal
+
+import (
+	"context"
+
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/pkg/errors"
+
+	"github.com/determined-ai/determined/master/internal/db"
+	"github.com/determined-ai/determined/proto/pkg/apiv1"
+	"github.com/determined-ai/determined/proto/pkg/taskv1"
+)
+
+func (a *apiServer) GetTask(
+	_ context.Context, req *apiv1.GetTaskRequest) (resp *apiv1.GetTaskResponse, err error) {
+	t := &taskv1.Task{}
+	switch err := a.m.db.QueryProto("get_task", t, req.TaskId); err {
+	case db.ErrNotFound:
```

Comment:
probably safer, in case someone wraps it one day
```suggestion
	switch err := a.m.db.QueryProto("get_task", t, req.TaskId); {
	case errors.Is(err, db.ErrNotFound):
```